src/lib/journey/callbacks/device-profile/device-profile.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+  import type { SelfSubmitFunction, StepMetadata } from '$journey/journey.interfaces';
+  import { FRDevice, type DeviceProfileCallback } from '@forgerock/javascript-sdk';
+  import type { Maybe } from '$lib/interfaces';
+  import type { ProfileConfigOptions } from '@forgerock/javascript-sdk/lib/fr-device/interfaces';
+  import Spinner from '$components/primitives/spinner/spinner.svelte';
+
+  export let spinner = false;
+  export let callback: Maybe<DeviceProfileCallback> = null;
+  export let callbackMetadata: Maybe<Record<string, any>> = null;
+  export let deviceConfig: ProfileConfigOptions = {};
```

Comment:
I don't think we'd want any device configuration coming from the step/stage component. If we had any custom configuration, we'd likely want that to be set in a Svelte Store when the Login Widget is configured initially (the whole `config.set()` method). Then, in this component, it would access that store for the config information. But, let's skip that for now. We can enhance this in a subsequent update.